src/less/button/button.less

```diff
@@ -295,24 +289,11 @@ a.fake-btn--tertiary[href] {
 }
 
 button.btn--tertiary[disabled],
-button.btn--tertiary[aria-disabled="true"]:not([aria-live="polite"][aria-disabled="true"]) {
-    .background-color-token(btn-tertiary-disabled-background-color, color-background-secondary);
-    .border-color-token(btn-tertiary-disabled-border-color, color-background-secondary);
-    .color-token(btn-tertiary-disabled-foreground-color, color-background-disabled);
-}
-
-a.fake-btn--tertiary:not([href]),
-a.fake-btn--tertiary[aria-disabled="true"] {
-    .background-color-token(btn-tertiary-disabled-background-color, color-background-secondary);
-    .border-color-token(btn-tertiary-disabled-border-color, color-background-secondary);
+button.btn--tertiary[aria-disabled="true"]:not([aria-live="polite"][aria-disabled="true"]),
+a.fake-btn--tertiary:not([href]) {
     .color-token(btn-tertiary-disabled-foreground-color, color-background-disabled);
 }
 
-button.btn--tertiary.btn--destructive,
-a.fake-btn--tertiary.fake-btn--destructive {
-    .color-token(btn-tertiary-destructive-foreground-color, color-foreground-attention);
-}
```

Comment:
It looks like destructive button has a different border:
<img width="879" alt="Screen Shot 2022-08-11 at 3 16 24 PM" src="https://user-images.githubusercontent.com/1755269/184251839-309d1f49-8100-45aa-8368-633193752ab2.png">